efficientdet/keras/inference.py

```diff
@@ -0,0 +1,364 @@
+    for i in range(len(imgs)):
+      driver.visualize(imgs[i], predictions[i])
+
+  Example 3: another way is to use SavedModel:
+
+    # step1: export a model.
+    driver = inference.ServingDriver('efficientdet-d0', '/tmp/efficientdet-d0')
+    driver.build()
+    driver.export('/tmp/saved_model_path')
+
+    # step2: Serve a model.
+    with tf.Session() as sess:
+      tf.saved_model.load(sess, ['serve'], self.saved_model_dir)
+      raw_images = []
+      for f in tf.io.gfile.glob('/tmp/images/*.jpg'):
+        raw_images.append(np.array(PIL.Image.open(f)))
+      detections = sess.run('detections:0', {'image_arrays:0': raw_images})
+      driver = inference.ServingDriver(
+        'efficientdet-d0', '/tmp/efficientdet-d0')
+      driver.visualize(raw_images[0], detections[0])
+      PIL.Image.fromarray(raw_images[0]).save(output_image_path)
+  """
+
+  def __init__(self,
+               model_name: Text,
+               ckpt_path: Text,
+               batch_size: int = 1,
+               min_score_thresh: float = None,
+               max_boxes_to_draw: float = None,
+               model_params: Dict[Text, Any] = None):
+    """Initialize the inference driver.
+
+    Args:
+      model_name: target model name, such as efficientdet-d0.
+      ckpt_path: checkpoint path, such as /tmp/efficientdet-d0/.
+      batch_size: batch size for inference.
+      use_xla: Whether run with xla optimization.
+      min_score_thresh: minimal score threshold for filtering predictions.
+      max_boxes_to_draw: the maximum number of boxes per image.
+      model_params: model parameters for overriding the config.
+    """
+    super().__init__()
+    self.model_name = model_name
+    self.ckpt_path = ckpt_path
+    self.batch_size = batch_size
+
+    self.params = hparams_config.get_detection_config(model_name).as_dict()
+
+    if model_params:
+      self.params.update(model_params)
+    self.params.update(dict(is_training_bn=False))
+    self.label_map = self.params.get('label_map', None)
+
+    self.model = None
+
+    self.min_score_thresh = min_score_thresh
+    self.max_boxes_to_draw = max_boxes_to_draw
+    mixed_precision = self.params.get('mixed_precision', None)
+    precision = utils.get_precision(
+        self.params.get('strategy', None), mixed_precision)
+    policy = tf.keras.mixed_precision.experimental.Policy(precision)
+    tf.keras.mixed_precision.experimental.set_policy(policy)
+
+  def build(self, params_override=None):
+    """Build model and restore checkpoints."""
+    params = copy.deepcopy(self.params)
+    if params_override:
+      params.update(params_override)
+    config = hparams_config.get_efficientdet_config(self.model_name)
+    config.override(params)
+    self.model = EfficientDetModel(config=config)
+    image_size = utils.parse_image_size(params['image_size'])
+    self.model.build((self.batch_size, *image_size, 3))
+    util_keras.restore_ckpt(self.model, self.ckpt_path, params['moving_average_decay'])
```

Comment:
try to load ckpt with restore_ckpt which is compatible with tf1.x ckpt and support load ema variables.